### PR TITLE
[Fixes] Update VM domainJoin parameter usage

### PR DESCRIPTION
**Pull request by jriekse5555**
Readme PR per request by Rainer Halanek PR related to bug I submitted (https://github.com/Azure/ResourceModules/issues/1595). Updated the domain join settings to the correct ones to domain join correctly. The previous ones in the readme do not work and would attempt to join a blank workgroup. Reference: https://docs.microsoft.com/en-us/azure/active-directory-domain-services/join-windows-vm-template

# Description

>Thank you for your contribution !

> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/Microsoft.Compute/virtualMachines/readme.md
Comment (ahmadabdalla) [inline modules/Microsoft.Compute/virtualMachines/readme.md:500]: The property on the vm module still references the old parameters names. Check "extensionDomainJoinPassword". The module and the readme need to be updated to retro fit the new changes.

**Review by ahmadabdalla — CHANGES_REQUESTED** [review]

File: modules/Microsoft.Compute/virtualMachines/readme.md
Comment (jriekse5555) [reply modules/Microsoft.Compute/virtualMachines/readme.md:500]: @ahmadabdalla Good catch. Updated.

File: modules/Microsoft.Compute/virtualMachines/deploy.bicep
Comment (ahmadabdalla) [inline modules/Microsoft.Compute/virtualMachines/deploy.bicep:169]: Thanks for fixing it @jriekse5555 . Also this same issue exists on the vm scale sets module. Can you update it there as well please? If you can also generate the readme again from the [script](https://github.com/Azure/ResourceModules/blob/main/utilities/tools/Set-ModuleReadMe.ps1) and would be great if you can attach a successful test pipeline status badge for both modules after the change.

**Review by ahmadabdalla — CHANGES_REQUESTED** [review]

File: modules/Microsoft.Compute/virtualMachines/deploy.bicep
Comment (jriekse5555) [reply modules/Microsoft.Compute/virtualMachines/deploy.bicep]: @ahmadabdalla Changes made in vm scale sets readme and module. Also tweaked the parameter descriptions in all four files from domainName to name. 

I haven't used the referenced script previously so would need more detailed instructions on how to leverage it.

Comment (ahmadabdalla) [reply modules/Microsoft.Compute/virtualMachines/deploy.bicep]: @jriekse5555 no problems:
If you load the script into your PowerShell session. Then try performing the below:

```pwsh
Set-ModuleReadMe -TemplateFilePath 'Path to your module directory\deploy.bicep'
```

So you need to run this script across both the VM and VMSS module, and it should replace the existing readme for you. Also just make sure to test the modules as well

Comment (jriekse5555) [reply modules/Microsoft.Compute/virtualMachines/deploy.bicep]: There appears to be a bit more to running that powershell script from the cloned copy I made of the PR fork (see powershell output below). With the parameter descriptions in the .bicep matching the readme (for both VM and VMSS) in theory it shouldn't change the readme file. 

Join-Path:
Line |
  34 |      . (Join-Path $PSScriptRoot 'helper/Merge-FileWithNewContent.ps1')
     |                   ~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'Path' because it is an empty string.
Split-Path:
Line |
  35 |      . (Join-Path (Split-Path $PSScriptRoot -Parent) 'pipelines' 'shar …
     |                               ~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'Path' because it is an empty string.
WARNING: A new Bicep release is available: v0.8.9. Upgrade now by running "az bicep upgrade".
Get-NestedResourceList:
Line |
  25 |      $RelevantResourceTypeObjects = Get-NestedResourceList $TemplateFi …
     |                                     ~~~~~~~~~~~~~~~~~~~~~~
     | The term 'Get-NestedResourceList' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
Merge-FileWithNewContent:
Line |
  56 |          $updatedFileContent = Merge-FileWithNewContent -oldContent $R …
     |                                ~~~~~~~~~~~~~~~~~~~~~~~~
     | The term 'Merge-FileWithNewContent' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
Set-ParametersSection:
Line |
 118 |          $readMeFileContent = Set-ParametersSection @inputObject
     |                                                     ~~~~~~~~~~~~
     | Cannot bind argument to parameter 'ReadMeFileContent' because it is null.
Set-OutputsSection:
Line |
 128 |          $readMeFileContent = Set-OutputsSection @inputObject
     |                                                  ~~~~~~~~~~~~
     | Cannot bind argument to parameter 'ReadMeFileContent' because it is null.
Set-DeploymentExamplesSection:
Line |
 140 |  …       $readMeFileContent = Set-DeploymentExamplesSection @inputObject
     |                                                             ~~~~~~~~~~~~
     | Cannot bind argument to parameter 'ReadMeFileContent' because it is null.
Set-TableOfContent:
Line |
 149 |          $readMeFileContent = Set-TableOfContent @inputObject
     |                                                  ~~~~~~~~~~~~
     | Cannot bind argument to parameter 'ReadMeFileContent' because it is null.

Comment (ahmadabdalla) [reply modules/Microsoft.Compute/virtualMachines/deploy.bicep]: @jriekse5555 will test this out and let you know

Comment (eriqua) [reply modules/Microsoft.Compute/virtualMachines/deploy.bicep]: Hey @jriekse5555, Which version of az cli do you have installed? To check
`az version`
If 2.37 you may have the issue tracked in https://github.com/Azure/ResourceModules/issues/1471. In that case try to upgrade to 2.38 with
`az upgrade`

Then rerun the script and let us know!

Comment (ahmadabdalla) [reply modules/Microsoft.Compute/virtualMachines/deploy.bicep]: @jriekse5555 I've replayed your changes locally and we do not need to run the readme script as it produced the same hash. However testing the module through pester is going to break when merged as we need to uplift the network interface module api version. See below

image

We need to fix that first before merging this change. I will raise an issue for that and then come back to this one.

Comment (ahmadabdalla) [reply modules/Microsoft.Compute/virtualMachines/deploy.bicep]: @jriekse5555 PR raised for #1722 to unblock this PR.

Comment (ahmadabdalla) [reply modules/Microsoft.Compute/virtualMachines/deploy.bicep]: PR Unlocked.

**Review by ahmadabdalla — APPROVED** [review]
Thanks for the contribution @jriekse5555

**Pull request merged**
